yt_dlp/extractor/zoom.py

```diff
@@ -51,27 +47,48 @@ def _real_extract(self, url):
                 raise ExtractorError(validation['errorMessage'], expected=True)
             webpage = self._download_webpage(url, play_id)
 
+        data: dict[str, Union[None, str, int, list, bool, dict[str, Any]]]
         data = self._parse_json(self._search_regex(
             r'(?s)window\.__data__\s*=\s*({.+?});',
             webpage, 'data'), play_id, js_to_json)
 
         subtitles = {}
-        for _type in ('transcript', 'cc'):
+        for _type in ('transcript', 'cc', 'chapter'):
             if data.get('%sUrl' % _type):
                 subtitles[_type] = [{
                     'url': urljoin(base_url, data['%sUrl' % _type]),
                     'ext': 'vtt',
                 }]
 
+        formats: list[dict[str, Union[None, str, int]]] = []
+
+        if data.get('shareMp4Url'):
+            formats.append({
+                'format_note': 'Screen share stream',
+                'url': str_or_none(data.get('shareMp4Url')),
+                'width': int_or_none(data.get('shareResolvtionsWidth')),
+                'height': int_or_none(data.get('shareResolvtionsHeight')),
+                'format_id': str_or_none(data.get('shareVideoId')),
+                'ext': 'mp4',
+            })
+
+        formats.append({
+            'format_note': 'Camera stream',
+            'url': str_or_none(data.get('viewMp4Url')),
+            'width': int_or_none(data.get('viewResolvtionsWidth')),
+            'height': int_or_none(data.get('viewResolvtionsHeight')),
+            'format_id': str_or_none(data.get('recordingId')),
+            'ext': 'mp4',
+            'filesize_approx': parse_filesize(data.get('fileSize')),
+        })
```

Comment:
the `filesize_approx` is only for the `view` and not the `share`, I checked and `share` does not have an equivalent
and its important for the `share` to be first, because then its the default when not specifying format and thus keeping backward compatibility (that the view is downloaded)
and the list comprehension does not check if the matching `url` exist, zoom always sends the field `shareMp4Url` for example, but its empty on videos without screen share

Reply:
And about adding the check to the second append
if the `viewMp4Url` then nothing else will, this is the main video and I haven't seen a zoom recording without it
the previous version did not check it, and it makes sense because this is the only thing it downloaded
I added the check for the `shareMp4Url` because it doesn't exist on videos that so not share screen
it makes more sense to throw an error if `viewMp4Url` doesn't exist

Reply:
There's a no_video_formats_error function afair, u can check for viewMp4Url and if it doesn't exist then raise that error

Reply:
the comprehension is overkill here imo. But do check for "if viewMp4Url".

> it makes more sense to throw an error if viewMp4Url doesn't exist

If `formats` is empty, the core code will raise appropriate error. No need to worry about t

Reply:
> There's a no_video_formats_error function afair, u can check for viewMp4Url and if it doesn't exist then raise that error

looked up for that but I didn't find it, is it documented somewhere?

Reply:
`raise_no_formats`?

Reply:
You can use `raise_no_formats`, but here since nothing else in the extractor breaks because of it. So the core code will handle raising the error